include/nil/blueprint/components/hashes/sha2/plonk/sha256_process.hpp

```diff
@@ -407,71 +474,91 @@ namespace nil {
                     const plonk_sha256_process<BlueprintFieldType, ArithmetizationParams> &component,
                     circuit<crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType, ArithmetizationParams>> &bp,
                     assignment<crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType, ArithmetizationParams>> &assignment,
-                    const std::map<std::string, std::size_t> &lookup_tables_indices
+                    const typename lookup_library<BlueprintFieldType>::left_reserved_type &lookup_tables_indices
                 ) {
                     std::array<std::size_t, 2> selectors;
                     using var = typename plonk_sha256_process<BlueprintFieldType, ArithmetizationParams>::var;
+                    using lookup_constraint = crypto3::zk::snark::plonk_lookup_constraint<BlueprintFieldType>;
+
+                    const std::array<var, 4> a_chunks_0_next = {
+                        var(component.W(1), +1), var(component.W(2), +1),
+                        var(component.W(3), +1), var(component.W(4), +1)};
+                    const std::array<var, 4> b_chunks_0 = {
+                        var(component.W(1), 0), var(component.W(2), 0),
+                        var(component.W(3), 0), var(component.W(4), 0)};
+                    const std::array<var, 4> b_chunks_1 = {
+                        var(component.W(1), -1), var(component.W(2), -1),
+                        var(component.W(3), -1), var(component.W(4), -1)};
+                    const std::array<std::array<var, 4>, 2> b_chunks = {b_chunks_0, b_chunks_1};
+
+                    const std::array<var, 4> sigma1_chunks_1 = {
+                        var(component.W(5), -1), var(component.W(6), -1),
+                        var(component.W(7), -1), var(component.W(8), -1)};
+                    const std::array<var, 4> a2_chunks = {
```

Comment:
var(component.W(8), 0) is integral_a2 , not a chunk

Reply:
Renamed here and in another place.